### PR TITLE
[KYUUBI #304] Validate specified host and port in arguments parse phase

### PR DESCRIPTION
**Pull request by turboFei**
turboFei [Closes #481](https://github.com/yaooqinn/kyuubi/pull/481)    Test Plan [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_

Validate specified host and port in arguments parse phase.

### _How was this patch tested?_
UT

### CODE REVIEW

File: kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala [hunk 1]
```diff
@@ -201,4 +201,35 @@ class KyuubiCtlArgumentsSuite extends KyuubiFunSuite {
     assert(opArgs3.verbose)
     assert(opArgs3.action == KyuubiCtlAction.HELP)
   }
+
+  test("test with unknown host") {
+    val args = Array(
+      "create", "server",
+      "-zk", zkAddress,
+      "-ns", namespace,
```
Comment (pan3793) [inline kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala:209]: I'm not sure if it's a good arg style. personally, I prefer
- single-letter Unix-style, i.e. `-v -p 1234` or `-v -p1234` or `-vp1234`
- all-word  GNU-style, i.e. `--verbose --port 1234` or `--verbose --port=1234`

But `java -version` dose not use that style

Comment (turboFei) [reply kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala:209]:
```
Arguments:
  --zkAddress, -zk            one of the zk ensemble address, using kyuubi-defaults/conf
                              if absent
  --namespace, -ns            the namespace, using kyuubi-defaults/conf if absent
  --host, -h                  hostname or IP address of a service
  --port, -p                  listening port of a service
  --version, -V               using the compiled KYUUBI_VERSION default, change it if the
                              active service is running in another
  --user, -u                  for engine service only, the user name this engine belong to
  --help, -I                  Show this help message and exit.
  --verbose, -v               Print additional debug output.
```

Comment (pan3793) [reply kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala:209]: Yes, I saw that, but we haven't release v1.2.0, if we have a better choice, the change is safe.

Comment (turboFei) [reply kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala:209]: do you mean that, align `--version` with `-v`?

Comment (pan3793) [reply kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala:209]: I suggest 

- change `--zkAddress, -zk` to `--zk-addr, -z`
- change `--help, -I` to `--help, -I`

to make it match more command line tool arg styles.

also cc @yaooqinn

Comment (turboFei) [reply kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala:209]: could you clear it for `--help`? @pan3793

Comment (yaooqinn) [reply kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala:209]: Well, I am not a fan of abbreviations, while it always confusing me such as `-h` here is for help or short. Shall we only use fully qualified parameters for v1.2.0?

Comment (pan3793) [reply kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala:209]: @turboFei sorry for typo, fixed

Comment (turboFei) [reply kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala:209]: sure, we can use full qualified parameters at first.

Comment (pan3793) [reply kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala:209]: @yaooqinn I'm OK with only GNU-style, especially in shell script, the single-letter arg is not clear sometimes.

Comment (turboFei) [reply kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala:209]: should we change `--zkAddress` to `--zk-addr`?
cc @yaooqinn

Comment (yaooqinn) [reply kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala:209]: on second thought, it's better to use `--zk-quorum` to match with `kyuubi.ha.zookeeper.quorum`

Comment (pan3793) [reply kyuubi-ctl/src/test/scala/org/apache/kyuubi/ctl/KyuubiCtlArgumentsSuite.scala:209]: +1 for `--zk-quorum`, please also change the text, `one of the zk ensemble address` is not properly here

**Comment by yaooqinn** [comment]
And also for the `CREATE` action, let's limit its use case only work for the `Kyuubi Server` that created in the same directory that the `kyuubi-ctl` runs, not for creating engines and a custom instance of a server. As an instance for exposing needs more verification than just a host and port, it's very likely for users to abuse. Consequently, we only allow `--zk-quorum` and namespace for `CREATE` action to expose the current server instance to another domain. Maybe that's enough for potential use cases that I can imagine so far.

**Comment by turboFei** [comment]
> And also for the `CREATE` action, let's limit its use case only work for the `Kyuubi Server` that created in the same directory that the `kyuubi-ctl` runs, not for creating engines and a custom instance of a server. As an instance for exposing needs more verification than just a host and port, it's very likely for users to abuse it. Consequently, we only allow `--zk-quorum` and namespace for `CREATE` action to expose the current server instance to another domain. Maybe that's enough for potential use cases that I can imagine so far.

ok， do you means that copy all the kyuubi server zk nodes to another namespace?

**Comment by yaooqinn** [comment]
> because kyuubi-ctl is only a client, and can not get the current server instance but get all the server instances under same namespace.

A I see, then let's not support engines for now

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master for v1.2.0

**Pull request closed without merging**